### PR TITLE
Don't fail silently when symlink source doesn't exist

### PR DESCRIPTION
**Pull request by nineinchnick**
Failing silently makes it hard to debug why some things stop working when changing paths.

There's a second commit formatting the whole file with [black](https://github.com/psf/black).

### CODE REVIEW

File: launcher/src/main/scripts/bin/launcher.py [hunk 1]
```diff
@@ -16,19 +16,23 @@
 from stat import S_ISLNK
 from time import sleep
 
-COMMANDS = ['run', 'start', 'stop', 'restart', 'kill', 'status']
+COMMANDS = ["run", "start", "stop", "restart", "kill", "status"]
```
Comment (findepi) [inline launcher/src/main/scripts/bin/launcher.py:19]: please move this to separate PR

File: launcher/src/main/scripts/bin/launcher.py [hunk 2]
```diff
@@ -169,8 +169,11 @@ def create_symlink(source, target):
     """Create a symlink, removing the target first if it is a symlink"""
     if symlink_exists(target):
         os.remove(target)
-    if exists(source):
-        os.symlink(source, target)
+    if not exists(source):
```
Comment (electrum) [inline launcher/src/main/scripts/bin/launcher.py:172]: I think this is the wrong behavior:

* we shouldn't remove the target if we're not going to create the symlink
* the `plugin` directory is optional, so this will break any application not using it

What is the purpose of this change?

Comment (nineinchnick) [reply launcher/src/main/scripts/bin/launcher.py:172]: The first point is a good one, this kind of validation should be done as the first thing in `create_symlink`.

As for the `plugin` dir, I did not realize it's optional. This change is intended to avoid a silent failure, where the `etc` option points to an invalid dir. The service would start but would fail to operate in an expected way, as there are a few paths pointing to etc in the data dir by default.

How about I'd add an explicit check before calling this for the etc dir only? Or even better, validate the etc dir option explicitly?

Comment (findepi) [reply launcher/src/main/scripts/bin/launcher.py:172]: > Or even better, validate the etc dir option explicitly?


sounds reasonable
